File: hw/ramos/u3/hw29Singleton/Java/AlertSystemDay/src/main/java/ec/edu/espe/alertsystemday/model/TaskRepository.java
package ec.edu.espe.alertsystemday.model;

/**
 *
 * @author Paulo Ramos
 */

import java.io.*;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

public class TaskRepository {

    private static final String FILE_PATH = "tasks.json";

    public List<Task> loadTasks() {
        List<Task> tasks = new ArrayList<>();

        File file = new File(FILE_PATH);
        if (!file.exists()) return tasks;

        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
            String line;
            while ((line = reader.readLine()) != null) {
                String[] parts = line.split(";");
                tasks.add(new Task(parts[0], LocalDate.parse(parts[1])));
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return tasks;
    }

    public void saveTasks(List<Task> tasks) {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(FILE_PATH))) {
            for (Task task : tasks) {
                writer.write(task.getName() + ";" + task.getDueDate());
                writer.newLine();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: hw/ramos/u3/hw29Singleton/python/AlertSystemPerDay/model/task_repository.py
from datetime import date
from model.task import Task


class TaskRepository:
    _file_path = "tasks.json"

    def load_tasks(self) -> list[Task]:
        tasks = []

        try:
            with open(self._file_path, "r") as file:
                for line in file:
                    parts = line.strip().split(";")
                    if len(parts) == 2:
                        task_name = parts[0]
                        task_date = date.fromisoformat(parts[1])
                        tasks.append(Task(task_name, task_date))
        except FileNotFoundError:
            pass
        except Exception as e:
            print("Error loading tasks:", e)

        return tasks

    def save_tasks(self, tasks: list[Task]):
        try:
            with open(self._file_path, "w") as file:
                for task in tasks:
                    file.write(f"{task.get_name()};{task.get_due_date()}\n")
        except Exception as e:
            print("Error saving tasks:", e)
